app/javascript/components/recipes/new/create-recipes.vue

```diff
@@ -70,32 +71,117 @@
         @click="create"
       />
     </div>
+
+    <base-modal
+      @ok="addIngredientstoRecipe"
+      @cancel="toggleAddIngredientModal"
+      v-if="showingAddIngredientModal"
+      :title="$t('msg.ingredients.add')"
+      :ok-button-label="$t('msg.add')"
+      :cancel-button-label="$t('msg.cancel')"
+    >
+      <div class="flex f-row w-full">
+        <div class="mr-10">
+          <search-ingredient-list 
+            :ingredients="my_ingredients"
+            @add-ingredient="addIngredient"
+          />
+        </div>
+        <selected-ingredients 
+          :elements="selectedIngredients"
+          :svg="{sixdots: true, cancel: false, menu_recipe: true}"
+          :input="false"
+          :placeholder="'-'"
+          @delete-ingredient="deleteIngredient"
+        /> 
+      </div>
+    </base-modal>
+
   </div>
 </template>
 
 <script>
 import List from './list.vue';
+import BaseModalComponent from '../../base/base-modal.vue';
+import SelectedIngredients from './selected-ingredients.vue';
+import SearchIngredientList from './search-ingredient-list.vue';
+import { getIngredients, postIngredient, deleteIngredient, editIngredient } from './../../../api/ingredients.js';
 
 export default {
   components: {
     List,
+    BaseModalComponent,
+    SelectedIngredients,
+    SearchIngredientList,
   },
   props: {
     objects: { type: Array, required: true },
   },
+  data() {
+    return {
+      showingAddIngredientModal: false,
+      selectedIngredients: [],
+      my_ingredients: [],
+      ingredients: [],
+    }
+  },
+  async created() {
+    try {
+      const response = await getIngredients();
+      this.my_ingredients = response.data.data.map((element) => ({
+        id: element.id,
+        ...element.attributes,
+      }));
+      this.successResponse(response);
+    } catch (error) {
+      this.errorResponse(error);
+    }
+  },
   methods: {
     create() {
       return null;
     },
-  },
-  computed: {
-    getIngredients() {
-      const newArray = [];
-      for (let i = 0; i < this.objects.length - 1; i++) {
-        newArray.push(this.objects[i]);
+    toggleAddIngredientModal(){
+      this.showingAddIngredientModal = !this.showingAddIngredientModal;
+    },
+    addIngredient(object){
+      var i;
+      for (i = 0; i < this.selectedIngredients.length; i++) {
+        if (this.selectedIngredients[i] == object){
+          return;
+        }
       }
+      this.selectedIngredients.push(object);
+    },
+    addIngredientstoRecipe(){
+      this.selectedIngredients.forEach(ingredient => {
+        var existent = false;
+        this.ingredients.forEach(existent_ingredient =>{
+          if (ingredient.id == JSON.parse(existent_ingredient)["id"]){
```

Comment:
⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>["id"] is better written in dot notation                                      dot-notation